lib/hls/hls_parser.js

```diff
@@ -2321,6 +2328,19 @@ shaka.hls.HlsParser = class {
           decodingInfos: [],
         };
 
+        if (audioStream) {
+          const mappedAudioVariantSet =
+              this.streamToVariants_.get(audioStream) || new Set();
+          mappedAudioVariantSet.add(audioStream);
+          this.streamToVariants_.set(audioStream, mappedAudioVariantSet);
+        }
+        if (videoStream) {
+          const mappedVideoVariantSet =
+              this.streamToVariants_.get(videoStream) || new Set();
+          mappedVideoVariantSet.add(videoStream);
```

Comment:
Oh, regarding my previous comment - I think it's caused by this line (`L2340` and `L2334`), as we're setting the value of this map to a set of `shaka.extern.Stream` objects, but they should be the set of `shaka.extern.Variant` objects:

`mappedVideoVariantSet.add(videoStream)` => `mappedVideoVariantSet.add(variant)`

Because currently, `streamToVariants_` has `<shaka.extern.Stream, Set<shaka.extern.Stream>>`.

Reply:
You are right about the maps.  I messed that up.  The fallout of that bug would be double-checking things that were already checked.

That doesn't explain a hang, but it's a definite bug.